### ISSUE TITLE
VERSION.txt is 0.4.2 but getLibSEDMLDottedVersion returns "1.0.1"

### ISSUE THREAD

**Issue by 0u812** [fbergmann/libSEDML#37]
I'm getting "1.0.1" from getLibSEDMLDottedVersion, which is different than what I expected. I don't know why. Any help would be appreciated. The VERSION.txt file in the the source dir says 0.4.2 (which matches the GitHub repo). Here is my generated common/libsedml-version.h file. I am on CentOS 5 w/ gcc 4.8.

```
/**
 * @file    libsedml-version.h
 * @brief   Define libSEDML version numbers for access from client software.
 *
 * <!--------------------------------------------------------------------------
 *
 * This file is part of libSEDML.  Please visit http://sed-ml.org for more
 * information about SED-ML. The latest version of libSEDML can be found on 
 * github: https://github.com/fbergmann/libSEDML/
 * 
 * 
 * Copyright (c) 2013-2014, Frank T. Bergmann  
 * All rights reserved.
 * 
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are met: 
 * 
 * 1. Redistributions of source code must retain the above copyright notice, this
 *    list of conditions and the following disclaimer. 
 * 2. Redistributions in binary form must reproduce the above copyright notice,
 *    this list of conditions and the following disclaimer in the documentation
 *    and/or other materials provided with the distribution. 
 * 
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
 * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
 * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
 * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
 * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
 * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
 * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
 * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 * 
 * ---------------------------------------------------------------------- -->
 */

#ifndef LIBSEDML_VERSION_H
#define LIBSEDML_VERSION_H 

#include <sedml/common/extern.h>


/**
 * LIBSEDML_DOTTED_VERSION:
 *
 * A version string of the form "1.2.3".
 */
#define LIBSEDML_DOTTED_VERSION	"1.0.1"


/**
 * LIBSEDML_VERSION:
 *
 * The version as an integer: version 1.2.3 becomes 10203.  Since the major
 * number comes first, the overall number will always increase when a new
 * libSEDML is released, making it easy to use less-than and greater-than
 * comparisons when testing versions numbers.
 */
#define LIBSEDML_VERSION		402


/**
 * LIBSEDML_VERSION_STRING:
 *
 * The numeric version as a string: version 1.2.3 becomes "10203".
 */
#define LIBSEDML_VERSION_STRING	"402"


LIBSEDML_CPP_NAMESPACE_BEGIN
BEGIN_C_DECLS

/**
 * Returns the version number of this copy of libSEDML as an integer.
 *
 * @return the libSEDML version as an integer; version 1.2.3 becomes 10203.
 */
LIBSEDML_EXTERN
int 
getLibSEDMLVersion ();


/**
 * Returns the version number of this copy of libSEDML as a string.
 *
 * @return the libSEDML version as a string; version 1.2.3 becomes
 * "1.2.3".
 *
 * @see getLibSEDMLVersionString()
 */
LIBSEDML_EXTERN
const char* 
getLibSEDMLDottedVersion ();


/**
 * Returns the version number of this copy of libSEDML as a string without
 * periods.
 *
 * @return the libSEDML version as a string: version 1.2.3 becomes "10203".
 *
 * @see getLibSEDMLDottedVersion()
 */
LIBSEDML_EXTERN
const char* 
getLibSEDMLVersionString ();


END_C_DECLS
LIBSEDML_CPP_NAMESPACE_END

#endif  /* LIBSEDML_VERSION_H */
```

**Comment by fbergmann** [comment]
could you please attach your cmakecache.txt file ... i'd like to figure out where the odd version number is coming from ... are you compiling other projects as well?

thanks
Frank

**Comment by 0u812** [comment]
Hi Frank. Here is the cmakecache. I am also building and linking to libSBML and libNuML. I noticed that the version for libNuML is [1.0.1](https://github.com/NuML/NuML/blob/master/libnuml/VERSION.txt) - could this be replacing the PACKAGE_VERSION CMake variable when [libNuML is imported](https://github.com/fbergmann/libSEDML/blob/master/CMakeLists.txt#L312)?
[CMakeCache.txt](https://github.com/fbergmann/libSEDML/files/699820/CMakeCache.txt)

**Comment by fbergmann** [comment]
Fixed with commit dd26182afde012b6cdb821c67dbeaeab64e97949, though I'm a bit puzzled as to why this happened in the first place ... clearly the cmake config of libsedml should override whatever else comes in from NuML ... 

anyways ... now neither numl nor libsedml use the package_version attribute for anything. And they both use unique variables this ought to do.
